pygeos/test/test_threading.py

```diff
@@ -0,0 +1,33 @@
+from concurrent.futures import ThreadPoolExecutor
+import pygeos
+import numpy as np
+import pytest
+
+
+def test_no_editing_when_multithreading():
+    a = pygeos.points(np.arange(1000), np.zeros(1000))
+    b = pygeos.points(np.zeros(1000), np.arange(1000))
+
+    def edit_arr(arr):
+        for i in range(arr.shape[0]):
+            arr[i] = None
+
+    with ThreadPoolExecutor(max_workers=2) as executor:
+        future = executor.submit(pygeos.distance, a, b)
+        with pytest.raises(ValueError):
+            executor.submit(edit_arr, a).result()
+        future.result()
+
+
+def test_editing_ok_when_single_worker():
```

Comment:
How does this work? It's not that we only set the writable flag if there are multiple threads?

Reply:
As you have only 1 worker, the two functions cannot be ran concurrently. I think this test is actually not necessary.